packages/react/src/hooks/useSwipe.ts

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+
+type UseSwipeOptions = {
+  minSwipeDistance?: number;
+};
+
+/**
+ * Simple implementation to detect horizontal swipe actions.
+ * Accepts callbacks for on right and left swipes.
+ * @example
+ * ```tsx
+ *  <div
+      onTouchStart={onTouchStart}
+      onTouchMove={onTouchMove}
+      onTouchEnd={onTouchEnd}
+    >
+ * @internal
+ * ```
+ */
+export function useSwipe(
```

Comment:
just as a suggestion: 
```
useSwipe(elementReference, onSwipe)
```

feels a lot simpler api wise and all the logic of registering events could be handled inside the hook. 
`onSwipe` would include a `direction` parameter, so we don't have to change the API if we need to add more swipe directions later on